community/kernel/src/main/java/org/neo4j/kernel/builtinprocs/BuiltInProcedures.java

```diff
@@ -283,6 +290,274 @@ public void resampleOutdatedIndexes()
+    public Stream<RelationshipResult> relationshipAutoIndexSeek( @Name( "key" ) String key,
+            @Name( "value" ) Object value ) throws ProcedureException
+    {
+        try ( Statement statement = tx.acquireStatement() )
+        {
+            ReadOperations readOperations = statement.readOperations();
+            LegacyIndexHits hits =
+                    readOperations.relationshipLegacyIndexGet( "relationship_auto_index", key, value, -1, -1 );
+            return toStream( hits, ( id ) -> new RelationshipResult( graphDatabaseAPI.getRelationshipById( id ) ) );
+        }
+        catch ( LegacyIndexNotFoundKernelException e )
+        {
+            // auto index will not exist if no relationships have been added that match the auto-index rules
+            return Stream.empty();
+        }
+    }
+
+    @Description( "Search relationship from automatic index. Replaces `START r=relationship:relationship_auto_index" +
+                  "('key:foo*')`" )
+    @Procedure( name = "db.index.auto.relationships", mode = READ )
+    public Stream<WeightedRelationshipResult> relationshipAutoIndexSearch( @Name( "query" ) Object query )
+            throws ProcedureException
+    {
+        try ( Statement statement = tx.acquireStatement() )
+        {
+            ReadOperations readOperations = statement.readOperations();
+            LegacyIndexHits hits =
+                    readOperations.relationshipLegacyIndexQuery( "relationship_auto_index", query, -1, -1 );
+            return toWeightedRelationshipResultStream( hits );
+        }
+        catch ( LegacyIndexNotFoundKernelException e )
+        {
+            // auto index will not exist if no relationships have been added that match the auto-index rules
+            return Stream.empty();
+        }
+    }
+
+    @Description( "Get or create a node manual index - YIELD type,name,config" )
+    @Procedure( name = "db.index.manual.forNodes", mode = WRITE )
+    public Stream<LegacyIndexInfo> nodeManualIndex( @Name( "indexName" ) String legacyIndexName )
+    {
+        IndexManager mgr = graphDatabaseAPI.index();
+        Index<Node> index = mgr.forNodes( legacyIndexName );
+        return Stream.of( new LegacyIndexInfo( "NODE", legacyIndexName, mgr.getConfiguration( index ) ) );
+    }
+
+    @Description( "Get or create a relationship manual index - YIELD type,name,config" )
+    @Procedure( name = "db.index.manual.forRelationships", mode = WRITE )
+    public Stream<LegacyIndexInfo> relationshipManualIndex( @Name( "indexName" ) String legacyIndexName )
+    {
+        IndexManager mgr = graphDatabaseAPI.index();
+        Index<Relationship> index = mgr.forRelationships( legacyIndexName );
+        return Stream.of( new LegacyIndexInfo( "RELATIONSHIP", legacyIndexName, mgr.getConfiguration( index ) ) );
+    }
+
+    @Description( "Check if a node manual index exists" )
+    @Procedure( name = "db.index.manual.exists.forNodes", mode = READ )
```

Comment:
It would seem easy enough to add a default parameter for index config, where the default is no config.